benchmarks/base.py

```diff
@@ -117,3 +121,40 @@ def _cleanup_tmp(self):
                 shutil.rmtree(os.path.join(tmp, path))
             elif path.startswith("tmp") and os.path.isfile(path):
                 os.remove(os.path.join(tmp, path))
+
+
+class BaseRemoteBench(BaseBench):
+
+    _remote_prefix = "s3://"
+    _remote_dir = "dvc-temp/dvc-bench"
+
+    def setup(self):
+        super().setup()
+
+        self.init_git()
+        self.init_dvc()
+
+        remote_url = (
+            self._remote_prefix
+            + self._remote_dir
+            + f"/temp-benchmarks-cache-{shortuuid.uuid()}"
+        )
+        self.dvc("remote", "add", "-d", "storage", remote_url)
+
+    def setup_data(self, template):
+        data_url = self._remote_dir + f"/tmp-benchmarks-data/{template}"
+        if self.fs.exists(data_url):
+            return data_url
+
+        local_url = f"_tmp_data_{template}"
+        self.gen(local_url, template)
+
+        self.fs.put(local_url, data_url, recursive=True)
```

Comment:
Isn't it `import-url` specific behaviour? Shouldn't we move it to `ImportToRemoteBench`?

Reply:
> Isn't it import-url specific behaviour?

We are going to need it on `dvc update --to-remote` + `dvc add -o ...` so that is why I added it as a boiler plate

Reply:
Understandable, lets keep it then - we can always refactor once we have some more grasp on which benchmarks are using it or not.